Applications/AdminUi/apps/admin_ui/lib/home/clients_overview/modals/change_client_secret_dialog.dart

```diff
@@ -132,7 +132,7 @@ class _ChangeClientSecretDialogState extends State<_ChangeClientSecretDialog> {
       return;
     }
 
-    _newClientSecretController.text = response.data.clientSecret;
+    _newClientSecretController.text = response.data.clientSecret!;
```

Comment:
So - there are two different responses. One containing clientSecret **always** and one containing it **never**. Why do you think the unsafe ! operator in combination with a nullable clientSecret is a better option than just two stricter types that handle exactly our usecases?

Reply:
It is not better at all, it is better to use more strictly types instead of unsafe ! . I see that I made mistake on that, I'll change it. Then what about the name of Clients should that be refactored?

Reply:
But the problem that I was facing before and forgot to mention in previous comment is now when I try to access the client details page I am getting an type error saying 
![image](https://github.com/user-attachments/assets/2288ce26-02f1-4ce3-8da6-f05209746b7c)

![image](https://github.com/user-attachments/assets/88c8f8b9-39ba-4eda-b3ce-f123ea21476a)

But the actual type of defaultTier is not a Object but only a simple String, that is why I used the first type and made critical error for clientSecret. My suggestion would be to create the third type just for Client Details?

Reply:
I am calling the correct route to get the specific Client.

Reply:
Right, and the route says string array. Why are you expecting a Map?

Reply:
This is the actual response when I get all Clients.
![image](https://github.com/user-attachments/assets/f229bf64-2d2f-444f-b0cd-7d4ecc8c0cf3)

Reply:
And that Is also why I abused extra for passing numberOfIdentities because that parameter is also not present in the endpoint for getting the specific client.

Reply:
But that's why there is a Client and a Clients.

@tnotheis Why does this have to be so hard? Could someone of the Backend guys rework this so that we don't have to do so much wichery here?

Reply:
> because that parameter is also not present in the endpoint for getting the specific client

the class says so?!

Reply:
the parameter will be available soon.

Reply:
This is the new output of the `GET /Clients` endpoint:
```json
{
	"result": [
		{
			"clientId": "test",
			"displayName": "test"
		}
	]
}
```

And this is the output of the `GET /Clients/{id}` endpoint:
```json
{
	"result": {
		"clientId": "test",
		"displayName": "test",
		"defaultTierId": "TIRadAOZG6HVRr54sQlP",
		"defaultTierName": "Basic",
		"createdAt": "2024-07-30T13:28:13.436Z",
		"maxIdentities": 9999
	}
}
```

If you need any specific fields added or removed, let me know.

Reply:
Why did you change Get clients? The only thing that should be changed was that get clients/id needs the additional field numberOfIdentities.

Reply:
This is how the problem was described:

> There is currently a problem in the Admin API: the response of `GET /Clients` has more properties than the response of `GET /Clients/{id}`, even though the latter are supposed to be the details, and the former the overview.

That's why I imagined that you only need some basic info when listing clients, and the exact info that is right now in listing clients should be in client details.

I will revert the changes and simply add the `numberOfIdentities` field.

Reply:
Looks good, it displays the info already:
![Screenshot 2024-08-01 185323](https://github.com/user-attachments/assets/3ae29fa0-3e58-4376-a17b-7b8f02773c17)